lib/chef/resource.rb

```diff
@@ -1071,14 +1072,37 @@ def self.default_action=(action_name)
     #
     # @return The Action class implementing the action
     #
-    def self.action(action, &recipe_block)
+    def self.action(action, description: nil, &recipe_block)
       action = action.to_sym
       declare_action_class
       action_class.action(action, &recipe_block)
       self.allowed_actions += [ action ]
+      # Accept any non-nil description, which will correctly override
+      # any specific inherited description.
+      action_descriptions[action] = description unless description.nil?
```

Comment:
I don't think the `unless` here is necessary. `action_descrptions[action]` will be `nil` if description is `nil`, and that's fine, ruby finds that falsey.

Reply:
This handles the case where if a base class has a description for `action`, it won't get replaced with `nil` if the inheriting resource overrides that action without a description. It may already be nil if the base class didn't describe it, but in case it's not we won't overwrite it with a nil.